src/sponge/pad.rs

```diff
@@ -35,6 +35,26 @@ where
     words
 }
 
+// Padding function for a singular scalar input 
+// to provide two outputs from a Poseidon hash
+pub fn pad_fixed_hash <T>(
+    capacity: T,
+    message: T,
+    pad_value: T,
+) -> Vec<T>
+where
+    T: Clone,
+{
+    // For a constant length hash, we always use a slice of width 5.
+    let width = 5;
+    let zero = pad_value;
+    let mut words = vec![zero; width];
```

Comment:
As I said, would not be difficult to turn that into a generic function for any number of inputs/outputs < WIDTH.

Anyway, for the time being maybe that's fine. Even I encourage you to try it.

Reply:
I would like to do this in the future when it is optimised for circuits.

Reply:
Doesn't have anything to do with circuits but ok

Reply:
See above for my train of thought